include/seqan3/io/sequence_file/format_fasta.hpp

```diff
@@ -329,47 +326,33 @@ class format_fasta
         if (options.fasta_blank_before_id)
             stream_it = ' ';
 
-        std::ranges::copy(id, stream_it);
-
+        stream_it.write_range(id);
         detail::write_eol(stream_it, options.add_carriage_return);
     }
 
     //!\brief Implementation of writing the sequence.
-    template <typename stream_it_t,
-              typename seq_type>
-    void write_seq(stream_it_t & stream_it,
-                   sequence_file_output_options const & options,
-                   seq_type && seq)
+    template <typename stream_it_t, typename seq_type>
+    void write_seq(stream_it_t & stream_it, sequence_file_output_options const & options, seq_type && seq)
     {
+        auto char_sequence = seq | views::to_char;
+
         if (options.fasta_letters_per_line > 0)
         {
-        #if SEQAN3_WORKAROUND_VIEW_PERFORMANCE
-            size_t count = 0;
-            for (auto c : seq)
+            /* Using `views::interleave` is probably the way to go but that needs performance-tuning.*/
+            auto it = std::ranges::begin(char_sequence);
+            auto end = std::ranges::end(char_sequence);
+            while (it != end)
             {
-                stream_it = to_char(c);
-                if (++count % options.fasta_letters_per_line == 0)
-                    detail::write_eol(stream_it, options.add_carriage_return);
-            }
-            if (count % options.fasta_letters_per_line != 0)
+                using subrange_t = ranges::subrange<decltype(it), decltype(it), ranges::subrange_kind::sized>;
+                /* Note: Calling detail::counting_next is suboptimal for non-random-access ranges.*/
```

Comment:
I think the current design is only suboptimal for sized non-random-access ranges. For all non-sized non-random-access ranges parsing outside eliminates the parsing inside (because the subrange being passed in is always sized now). So the solution is really good for 98% of the cases, I think.